File: apps/fluster/lib/data_models/setting/setting_section.dart
import 'package:fluster/data_models/setting/setting_abstract.dart';

class SettingSection<T extends SettingAbstract> {
  final String? label;
  final String? subtitle;
  final List<SettingAbstract> items;
  const SettingSection({this.label, this.subtitle, required this.items});
}

File: apps/fluster/src-tauri/src/core/models/settings/setting_section.rs
use serde::{Deserialize, Serialize};

use super::traits::setting_trait::SettingItem;

#[derive(Clone, Serialize, Deserialize)]
pub struct SettingSection {
    pub label: Option<String>,
    pub subtitle: Option<String>,
    pub items: Vec<SettingItem>,
}
